plumbing/protocol/packp/srvresp.go

```diff
@@ -101,6 +101,11 @@ func (r *ServerResponse) decodeLine(line []byte) error {
 		return fmt.Errorf("unexpected flush")
 	}
 
+	err := fmt.Errorf("unexpected content %q", string(line))
+	if len(line) < 3 {
+		return err
```

Comment:
```suggestion
	if l := len(line); l < 3 {
		return fmt.Errorf("unexpected content %q", string(line))
```

maybe?

Reply:
That would break line 117... 

Now the current change will result in the unnecessary stringification of `line` which depending on the length of line this may or may not be a good idea.

It's probably better to do:

```go
func (r *ServerResponse) decodeLine(line []byte) error {
	if len(line) == 0 {
		return fmt.Errorf("unexpected flush")
	}

	if len(line) >= 3 {
		if bytes.Equal(line[0:3], ack) {
			return r.decodeACKLine(line)
		}
		if bytes.Equal(line[0:3], nak) {
			return nil
		}
	}
	return fmt.Errorf("unexpected content %q", string(line))
}
```

Or duplicate the `fmt.Errorf` statement, return a slightly different error e.g. `fmt.Errorf("too short content: %q", string(line))`, or test the len twice.

Reply:
nice shout, changed for the latter approach. PTAL